operator/clusterutilities_test.go

```diff
@@ -0,0 +1,170 @@
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/crunchydata/postgres-operator/config"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+func mockSetupContainers(values map[string]struct {
+	name  string
+	image string
+}) []v1.Container {
+	containers := []v1.Container{}
+
+	for _, value := range values {
+		container := v1.Container{
+			Name:  value.name,
+			Image: value.image,
+		}
+
+		containers = append(containers, container)
+	}
+
+	return containers
+}
+
+func TestOverrideClusterContainerImages(t *testing.T) {
+
+	containerDefaults := map[string]struct {
+		name  string
+		image string
+	}{
+		"database":   {name: "database", image: config.CONTAINER_IMAGE_CRUNCHY_POSTGRES_HA},
+		"crunchyadm": {name: "crunchyadm", image: config.CONTAINER_IMAGE_CRUNCHY_ADMIN},
+		"collect":    {name: "collect", image: config.CONTAINER_IMAGE_CRUNCHY_COLLECT},
+		"pgbadger":   {name: "pgbadger", image: config.CONTAINER_IMAGE_CRUNCHY_PGBADGER},
+		"future":     {name: "future", image: "crunchy-future"},
+	}
+
+	t.Run("no override", func(t *testing.T) {
+		containers := mockSetupContainers(containerDefaults)
+
+		OverrideClusterContainerImages(containers)
+
+		for _, container := range containers {
+			containerDefault, ok := containerDefaults[container.Name]
+
+			if !ok {
+				t.Errorf("could not find container %q", container.Name)
+				return
+			}
+
+			if containerDefault.image != container.Image {
+				t.Errorf("image overwritten when it should not have been. expected %q actual %q",
+					containerDefault.image, container.Image)
+			}
+		}
+	})
+
+	// test overriding each container and ensure that it takes in the contianer
+	// slice. Skip the "future" container, that will be in an upcoming test
+	for name, defaults := range containerDefaults {
+		if name == "future" {
+			continue
+		}
+
+		t.Run(fmt.Sprintf("override %s", name), func(t *testing.T) {
+			// override the struct that contains the value
+			ContainerImageOverrides[defaults.image] = "overriden"
+			containers := mockSetupContainers(containerDefaults)
+
+			OverrideClusterContainerImages(containers)
+
+			// determine if this container is overriden
+			for _, container := range containers {
+				containerDefault, ok := containerDefaults[container.Name]
+
+				if !ok {
+					t.Errorf("could not find container %q", container.Name)
+					return
+				}
+
+				if containerDefault.name == name && containerDefault.image == container.Image {
+					t.Errorf("container %q not overwritten. image name is %q",
+						containerDefault.name, container.Image)
+				}
+			}
+			// unoverride at the end of the test
+			ContainerImageOverrides[defaults.image] = ""
```

Comment:
Would just deleting the entry from the map at the end of the test make sense here, e.g.:
```
delete(ContainerImageOverrides, ContainerImageOverrides[defaults.image])
```